runtime/block_reduction.cu

```diff
@@ -111,3 +111,134 @@ __device__ void blockReduce(
       read_write_pred,
       init_val);
 }
+
+// Each thread in the iteration dimension processes N elements
+// Typical usage is in outer reduction where the iteration dimension
+// is parallelized by vectorized loads, bidmx. The reduction dimension
+// is parallelized by bdimy. This function works as follows:
+// (1) Each thread vectorized loads N elements from input register array to
+// smem. (2) do N * bdimx parallel reductions in smem.
+template <
+    bool X_REDUCE,
+    bool Y_REDUCE,
+    bool Z_REDUCE,
+    bool Aligned,
+    int N, // Number of elements per input array
+    typename T,
+    typename Func>
+__device__ void blockIterGroupedReduce(
```

Comment:
Could this be unified with `blockReduce` instead of having a separate name? It seems like it's a pretty straight-forward extension. Is there any part of this that you'd expect a slowdown for `N=1` compared to `blockReduce`?

Reply:
make sense. will consider merge them after `gridIterGoupedReduce`